web/src/js/components/ContentView/ContentViews.jsx

```diff
@@ -18,43 +20,69 @@ function ViewImage({ flow, message }) {
     )
 }
 
-
-ViewRaw.matches = () => true
-ViewRaw.propTypes = {
+Edit.propTypes = {
     content: React.PropTypes.string.isRequired,
 }
-function ViewRaw({ content, readonly, onChange }) {
-    return readonly ? <pre>{content}</pre> : <CodeEditor content={content} onChange={onChange}/>
+
+function Edit({ content, onChange }) {
+    return <CodeEditor content={content} onChange={onChange}/>
 }
-ViewRaw = ContentLoader(ViewRaw)
+Edit = ContentLoader(Edit)
 
+class ViewServer extends Component {
+    static defaultProps = {
+        maxLines: 80,
+    }
 
-const isJSON = /^application\/json$/i
-ViewJSON.matches = msg => isJSON.test(MessageUtils.getContentType(msg))
-ViewJSON.propTypes = {
-    content: React.PropTypes.string.isRequired,
-}
-function ViewJSON({ content }) {
-    let json = content
-    try {
-        json = JSON.stringify(JSON.parse(content), null, 2);
-    } catch (e) {
-        // @noop
+    componentWillMount(){
```

Comment:
This should be done in the constructor, otherwise we'll have to render twice if I understand things correctly.